port/port_posix.h

```diff
@@ -202,7 +202,14 @@ extern void *cacheline_aligned_alloc(size_t size);
 
 extern void cacheline_aligned_free(void *memblock);
 
+#if defined(__aarch64__)
+//  __builtin_prefetch(..., 1) turns into a prefetch into prfm pldl3keep. On
+//  arm64 get these into the L1 and L2
+#define PREFETCH(addr, rw, locality) \
+  __builtin_prefetch(addr, rw, locality == 1 ? 3 : locality)
```

Comment:
The mapping still feels odd to me:
0->0, 1->3, 2->2, 3->3.

It's not monotonic increasing.

Reply:
0 is special, it means non-temporal.  I could find any uses in the code of 2, so one way would be to just write:
```locality != 0 ? 3 : 0``` which would mean 0->0; 1,2,3->3

Reply:
> 

I think it feels better. If we do 1->2, what the performance would be?

Reply:
We're still leaving a fair bit (5%) on the table...
```
           1 -> 2     1 -> 3
----------------------------
Graviton3   +10%        +15%
M1          +10%        +10%
```

To be actually used the data is going to end up in the L1 cache, so as long as we're not over prefetching (e.g. prefetching data we have a low chance of utilizing) and the use is going to be "soon" then the line is going to be making it's way into the L1 shortly and we might as well help it along.

Reply:
OK. Either 1->2, or 1,2->3 sounds good to me.

Reply:
ack, updated.